### PR TITLE
Update Logging article, log4j2

### PR DESCRIPTION
**Pull request by AngelosF**
Closes https://github.com/openhab/openhab-docs/issues/476 (yes, 1 year later 👍 )

Signed-off-by: Angelos Fountoulakis agf@wired-net.gr (github: AngelosF)

### CODE REVIEW

**Comment by ScottKinSF** [comment]
Would be getting too detailed to add a link to the [log4j PatternLayout reference page](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html)? I recently wanted to add thread IDs to my logs. That page provided the information I needed to do exactly that.

**Comment by AngelosF** [comment]
> Would be getting too detailed to add a link to the [log4j PatternLayout reference page](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html)? I recently wanted to add thread IDs to my logs. That page provided the information I needed to do exactly that.

I believe that it's a bit too advanced and people who want to learn more about conversion patterns will google them (like you).  I found also the log4j2 version here: https://logging.apache.org/log4j/2.x/manual/layouts.html (section `Pattern Layout`)

File: administration/logging.md [hunk 1]
```diff
@@ -61,7 +65,7 @@ The config file for logging is `org.ops4j.pax.logging.cfg` located in the `userd
 
 ## Defining what to log
 
-In order to see the messages, logging needs to activated defining what should be logged and in which detail. This can be done in Karaf using the following console command:
+In order to see the messages, logging needs to activated, defining what should be logged and in which detail. This can be done in Karaf using the following console command:
```
Comment (Confectrician) [inline administration/logging.md]: Please split up to one line per Sentence.

File: administration/logging.md [hunk 2]
```diff
@@ -71,15 +75,18 @@ The **what** is defined by `package.subpackage` and is in most cases a binding (
...
+| ERROR               | 200           | When an error in the application, possibly recoverable                                  |
+| WARN                | 300           | When an event that might possible lead to an error                                      |
+| INFO                | 400           | When an event for informational purposes                                                |
+| DEBUG               | 500           | When a general debugging event required                                                 |
+| TRACE               | 600           | When a fine grained debug message, typically capturing the flow through the application |
+| ALL                 | MAX           | When all events should be logged                                                        |
+
+The levels build a hierarchy with **ERROR** logging critical messages only and **DEBUG** logging nearly everything. **DEBUG** combines all logs from weight 100 to 500, while **TRACE** adds further messages in addition to what **DEBUG** displays.  **ALL** includes every log level from weight 100 to 600.
```
Comment (Confectrician) [inline administration/logging.md]: Please split up to one line per Sentence.

File: administration/logging.md [hunk 3]
```diff
@@ -88,7 +95,7 @@ Following example sets the logging for the Z-Wave binding to **DEBUG**
 log:set DEBUG org.openhab.binding.zwave
 ```
 
-Note that the log levels set using the `log:set` commands are not persistent and will be lost upon restart. To configure those in a persistent way, the commands have to be added to the [configuration file](#config-file).
+Note that the log levels set using the `log:set` commands are persistent and will be applied upon restart.  To modify the stored log levels, use the console or edit the [configuration file](#config-file).
```
Comment (Confectrician) [inline administration/logging.md]: Please split up to one line per Sentence.

File: administration/logging.md (hunk 1 above)
Comment (AngelosF) [reply administration/logging.md]: Done 👍

File: administration/logging.md (hunk 2 above)
Comment (AngelosF) [reply administration/logging.md]: Done 👍

File: administration/logging.md (hunk 3 above)
Comment (AngelosF) [reply administration/logging.md]: Done 👍

**Comment by bgilmer77** [comment]
@AngelosF This looks very good to me.  Thanks for working on it.

**Pull request merged**